File: iosApp/iosAppTests/offer/TestOfferService.swift
//
//  TestOfferService.swift
//  iosAppTests
//
//  Created by jimmyt on 9/3/22.
//  Copyright © 2022 orgName. All rights reserved.
//

@testable import iosApp

/**
 A basic `OfferNotifiable` implementation used to satisfy `OfferService`'s `offerService` dependency for testing non-offer-related code.
 */
class TestOfferService: OfferNotifiable {
    /**
     Does nothing, required to adopt `OfferNotifiable`. Should not be used.
     */
    func handleOfferOpenedEvent(_ event: OfferOpenedEvent) {}
    /**
     Does nothing, required to adopt `OfferNotifiable`. Should not be used.
     */
    func handleOfferEditedEvent(_ event: OfferEditedEvent) {}
    /**
     Does nothing, required to adopt `OfferNotifiable`. Should not be used.
     */
    func handleOfferCanceledEvent(_ event: OfferCanceledEvent) {}
    /**
     Does nothing, required to adopt `OfferNotifiable`. Should not be used.
     */
    func handleOfferTakenEvent(_ event: OfferTakenEvent) {}
    /**
     Does nothing, required to adopt `OfferNotifiable`. Should not be used.
     */
    func handleServiceFeeRateChangedEvent(_ event: ServiceFeeRateChangedEvent) {}
}

File: androidApp/src/test/java/com/commuto/interfacemobile/android/offer/TestOfferService.kt
package com.commuto.interfacemobile.android.offer

import com.commuto.interfacemobile.android.blockchain.BlockchainService
import com.commuto.interfacemobile.android.blockchain.BlockchainTransaction
import com.commuto.interfacemobile.android.blockchain.BlockchainTransactionException
import com.commuto.interfacemobile.android.blockchain.events.commutoswap.*
import com.commuto.interfacemobile.android.blockchain.events.erc20.ApprovalEvent

/**
 * A basic [OfferNotifiable] implementation used to satisfy [BlockchainService]'s offerService dependency for testing
 * non-offer-related code.
 */
class TestOfferService: OfferNotifiable {
    override suspend fun handleFailedTransaction(
        transaction: BlockchainTransaction,
        exception: BlockchainTransactionException
    ) {}
    override suspend fun handleTokenTransferApprovalEvent(event: ApprovalEvent) {}
    override suspend fun handleOfferOpenedEvent(event: OfferOpenedEvent) {}
    override suspend fun handleOfferEditedEvent(event: OfferEditedEvent) {}
    override suspend fun handleOfferCanceledEvent(event: OfferCanceledEvent) {}
    override suspend fun handleOfferTakenEvent(event: OfferTakenEvent) {}
    override suspend fun handleServiceFeeRateChangedEvent(event: ServiceFeeRateChangedEvent) {}
}